Classes/KIFUITestActor.m

```diff
@@ -149,8 +149,26 @@ - (void)waitForAnimationsToFinishWithTimeout:(NSTimeInterval)timeout {
         __block BOOL runningAnimationFound = false;
         for (UIWindow *window in [UIApplication sharedApplication].windowsWithKeyWindow) {
             [window performBlockOnDescendentViews:^(UIView *view, BOOL *stop) {
-                BOOL isViewVisible = [view isVisibleInViewHierarchy];   // do not wait for animatinos of views that aren't visible
-                BOOL hasAnimation = view.layer.animationKeys.count != 0 && ![view.layer.animationKeys isEqualToArray:@[@"_UIParallaxMotionEffect"]];    // explicitly exclude _UIParallaxMotionEffect as it is used in alertviews, and we don't want every alertview to be paused
+                BOOL isViewVisible = [view isVisibleInViewHierarchy];   // do not wait for animations of views that aren't visible
+                BOOL(^layerAnimationTest)(CALayer *layer);
+                __block __weak BOOL(^weakLayerAnimationTest)(CALayer *layer); // There was no retain cycle anyway, but this is the easiest way to keep the compiler happy
+                layerAnimationTest = ^(CALayer *layer) {
```

Comment:
Instead of using a block, perhaps it would be better to extend CALayer with a `hasAnimations` method.